pkg/apis/core/validation/validation.go

```diff
@@ -6219,33 +6243,116 @@ func validatePodResizeContainerOrdering(newPod, oldPod *core.Pod, specPath *fiel
-func dropCPUMemoryResourcesFromContainer(container *core.Container, oldPodSpecContainer *core.Container) {
-	dropCPUMemoryUpdates := func(resourceList, oldResourceList core.ResourceList) core.ResourceList {
-		if oldResourceList == nil {
-			return nil
-		}
-		var mungedResourceList core.ResourceList
-		if resourceList == nil {
-			mungedResourceList = make(core.ResourceList)
-		} else {
-			mungedResourceList = resourceList.DeepCopy()
+// isPodLevelResourcesResizeRequestSupported checks whether the pod supports resize requests
+// by verifying if the pod is running and has pod-level resources populated.
+func isPodLevelResourcesResizeRequestSupported(pod core.Pod) bool {
+	// If pod is not running, we cannot tell whether the node supports resize
+	// at this point, so we assume it does.
+	if pod.Status.Phase != core.PodRunning {
+		return true
+	}
+
+	// Pod is running - check if pod-level resources are populated
+	// which indicates the node/kubelet supports in-place vertical scaling.
+	return pod.Status.Resources != nil
+}
+
+func validatePodLevelResourcesResize(newPod, oldPod *core.Pod, specPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	var allErrs field.ErrorList
+	if !isPodLevelResourcesResizeRequestSupported(*oldPod) {
+		allErrs = append(allErrs, field.Forbidden(specPath, "Pod running on node without support for resize"))
+	}
+
+	if resourcesRemoved(newPod.Spec.Resources.Requests, oldPod.Spec.Resources.Requests) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("resources").Child("requests"), "resource requests cannot be removed"))
+	}
+
+	if resourcesRemoved(newPod.Spec.Resources.Limits, oldPod.Spec.Resources.Limits) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("resources").Child("limits"), "resource limits cannot be removed"))
+	}
+
+	var memRestartPolicy core.ResourceResizeRestartPolicy
+
+	for _, container := range newPod.Spec.InitContainers {
+		for _, policy := range container.ResizePolicy {
+			if policy.ResourceName == core.ResourceMemory {
+				memRestartPolicy = policy.RestartPolicy
+				break
+			}
 		}
-		delete(mungedResourceList, core.ResourceCPU)
-		delete(mungedResourceList, core.ResourceMemory)
-		if cpu, found := oldResourceList[core.ResourceCPU]; found {
-			mungedResourceList[core.ResourceCPU] = cpu
+
+	}
+	for _, container := range newPod.Spec.Containers {
+		for _, policy := range container.ResizePolicy {
+			if policy.ResourceName == core.ResourceMemory {
+				memRestartPolicy = policy.RestartPolicy
+				break
+			}
 		}
-		if mem, found := oldResourceList[core.ResourceMemory]; found {
-			mungedResourceList[core.ResourceMemory] = mem
+	}
+
+	allErrs = append(allErrs, validateMemoryResize(memRestartPolicy, newPod.Spec, oldPod.Spec)...)
+
+	return allErrs
+}
+
+func validateMemoryResize(memRestartPolicy core.ResourceResizeRestartPolicy, newPodSpec, oldPodSpec core.PodSpec) field.ErrorList {
```

Comment:
We allow memory limit decrease for container-level IPPR now. I don't see it written in the KEP that we will forbid this for PLR IPPR, so I think we want to support it for parity unless there is some technical constraint that I am not aware of? 

Non-blocking for alpha I think, but at minimum we should add a TODO and revisit it in beta